### PR TITLE
First set of changes related to serving localised content on web.dev

### PR DESCRIPTION
**Pull request by devnook**
This is the first part of changes related to serving localised content on web.dev. 

It is mostly dealing with the fact, that we relied on post's slug to be unique, which in the world of translated content is not true (e.g. we have /en/some-post/index.html and /pl/some-post/index.html, both have slug "some-post").

1. I had to change memoization of posts in collection.all to use url (language+slug) instead of slug only.
2. I changed permalins to remove hardcoded /en/ language in path. I left /en/ hardcoded on sitemap and algolia urls, as they are service urls and not user-facing ones.
This said, I'd need to look into localisation of search&algolia separately.
3. I moved /handbook/... to /..., like all other urls. Is there a reason it has to stay in /handbook/?
4. I added a /pl/ folder so that we can test this language setup.

### CODE REVIEW

**Comment by devnook** [comment]
One more thing, in case this is not clear from the first sight:
5. The new permalinks create a flat structure in /dist, so e.g. instead of:
 `/dist/en/accessible/some-post/index.html` and
 `/dist/en/accessible/index.html`
we have now :
 `/dist/en/accessible/index.html`
 `/dist/en/some-post/index.html`

**Comment by robdodson** [comment]
I wanted to mention that if we didn't want to have to pass the `lang` argument everywhere, then we could maybe switch from using Eleventy shortcodes to using nunjucks includes. Includes have access to the global data context, so the `lang` property would cascade down to them from the `/en` and `/pl` directory data files. However, nunjucks includes are potentially harder to work with because you have to write them as jinja templates and you lose out on a lot of niceness and debuggability of javascript templates.

But I wanted to mention it as a potential solution if we feel like passing `lang` is too annoying.

File: src/site/_filters/find-by-url.js [hunk 1]
```diff
@@ -35,38 +35,40 @@ const memoize = (collection) => {
 
   memo = {};
```
[the remark this answers is not in the document]
Comment (devnook) [reply src/site/_filters/find-by-url.js]: done

File: src/site/_includes/components/CodelabsCallout.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ module.exports = (slugs) => {
   // Coerce slugs to Array just in case someone pasted in a single slug string.
   slugs = slugs instanceof Array ? slugs : [slugs];
 
-  const codelabs = slugs.map((slug) => findBySlug(slug));
+  const codelabs = slugs.map((slug) => findByUrl(`/en/${slug}/`));
```
[the remark this answers is not in the document]
Comment (devnook) [reply src/site/_includes/components/CodelabsCallout.js]: done

**Comment by devnook** [comment]
We could potentially remove a lot of lang passing by using urls instead of slugs in out 11ty datafiles for paths (there is no need to rewrite slug back to language-aware url then). This would on the other hand make it more troublesome to curate 11ty path files for each language.

**Comment by robdodson** [comment]
> We could potentially remove a lot of lang passing by using urls instead of slugs in out 11ty datafiles for paths (there is no need to rewrite slug back to language-aware url then). This would on the other hand make it more troublesome to curate 11ty path files for each language.

That's interesting...

I guess since 11ty data files can run as a function before exporting their object, you could have them prepend the lang to the start of the urls before they export the final object.

**Comment by devnook** [comment]
I gave it a try and prepending language to slugs in 11tydata files proved not practical. We use mostly canonical urls throughout the website, so we'd end up stripping language in roughly as many places as we avoid passing lang, so no point.

**Comment by devnook** [comment]
PTAL

**Pull request merged**